paddle/fluid/framework/grad_op_desc_maker.h

```diff
@@ -55,7 +55,10 @@ class GradOpDescMakerBase {
                    std::back_inserter(ret_val),
                    [this](const std::string& fwd_var_name) -> std::string {
                      auto g_name = GradVarName(fwd_var_name);
-                     if (no_grad_set_.count(g_name)) {
+                     if (no_grad_set_.empty()) {
```

Comment:
`no_grad_set_.empty() || !no_grad_set_.count(g_name)`

and remove the else below?

Reply:
Done